control_panel_api/aws.py

```diff
@@ -0,0 +1,85 @@
+import json
+
+import boto3
+
+
+def create_bucket(name, **kwargs):
+    boto3.client('s3').create_bucket(
+        Bucket=name,
+        ACL=kwargs.get('acl', ''),
+        CreateBucketConfiguration={'LocationConstraint': kwargs.get('region', '')},
+    )
+
+
+def put_bucket_logging(name, **kwargs):
+    boto3.client('s3').put_bucket_logging(
+        Bucket=name,
+        BucketLoggingStatus={
+            'LoggingEnabled': {
+                'TargetBucket': kwargs.get('target_bucket', ''),
+                'TargetPrefix': kwargs.get('target_prefix', '')
+            }
+        }
+    )
+
+
+def create_policy(name, readwrite=False):
+    boto3.client("iam").create_policy(
+        PolicyName=name,
+        PolicyDocument=json.dumps(get_policy_document(name, readwrite)),
```

Comment:
There is a problem here: The `name` here is the name of the policy (coming from the `services` module - however the `get_policy_document()` function use it as `bucket_name` (e.g. to build the `bucket_arn` and determine on which S3 bucket you can perform these operations).

This mean that as it stands the IAM policy document would not work because would contain things like `"Resource": ["arn:aws:s3:::dev-mybucket-readonly"]`. The resource/bucket arn should just be `arn:aws:s3:::dev-mybucket`

Reply:
As a related comment, I think we should try to use variable names which are a bit more explicit, e.g. `bucket_name` or `policy_name` whenever possible to avoid this kind of confusions.

Although to be fair, I think if the function (`create_policy`) is clearly operating on a certain thing (`policy`) it would be OK to omit the `_policy` suffix, as in this case.